### PR TITLE
DVDVideoCodecDRMPRIME: remove depreciated ffmpeg function

### PR DESCRIPTION
**Pull request by lrusak**
`av_codec_next` is depreciated in ffmpeg 4.0 see [here](https://www.ffmpeg.org/doxygen/trunk/group__lavc__core.html#ga424c2b3c23cc41f4ef93ef2034f5c901)

### CODE REVIEW

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp [hunk 1]
```diff
@@ -173,16 +173,18 @@ void CDVDVideoCodecDRMPRIME::Register()
...
+  while ((codec = av_codec_iterate(&i)))
   {
     if (av_codec_is_decoder(codec) && codec->id == hints.codec && codec->pix_fmts)
     {
       const AVPixelFormat* fmt = codec->pix_fmts;
       while (*fmt != AV_PIX_FMT_NONE)
       {
         if (*fmt == AV_PIX_FMT_DRM_PRIME)
-          return codec;
+          return const_cast<AVCodec*>(codec);
```
Comment (notspiff) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp]: shouldn't we fix the constness instead ?

Comment (lrusak) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/DVDVideoCodecDRMPRIME.cpp]: fixed

**Comment by Kwiboo** [comment]
@lrusak I would like to change `FindDecoder` to make use of the new `avcodec_get_hw_config` api in addition to just dropping the `av_codec_next` call. I will send a PR to your tree later today.

**Comment by Kwiboo** [comment]
@lrusak please drop https://github.com/xbmc/xbmc/pull/13813/commits/8a2e8bcddfa6793f5a5f5e6a7fac4bd79a3a8a1c and pick top 3 commits from https://github.com/Kwiboo/plex-home-theater/commits/drmprime-ffmpeg4 as a replacement, adds hwdevice support and a get_format callback, this should make the codec selection and configuration rather future proof, tested with rkmpp decoder and rkvpu hwaccel

**Comment by lrusak** [comment]
@Kwiboo really great! thanks for doing the `get_format` callback!!

**Comment by lrusak** [comment]
tested working on RPi3b+ with v4l2 :+1:

**Pull request merged**